src/main/kotlin/com/openlattice/launchpad/configuration/IntegrationRunner.kt

```diff
@@ -61,36 +66,78 @@ class IntegrationRunner {
             id.toString()
         }
 
+        fun configureOrGetSparkSession(integrationConfiguration: IntegrationConfiguration): SparkSession {
+            val session = SparkSession.builder()
+                    .master("local[${Runtime.getRuntime().availableProcessors()}]")
+                    .appName("integration")
+            if ( integrationConfiguration.awsConfig.isPresent ){
+                val config = DefaultAWSCredentialsProviderChain.getInstance().credentials
+                val manualConfig = integrationConfiguration.awsConfig.get()
+                session
+                        .config("fs.s3a.access.key", config.awsAccessKeyId)
+                        .config("fs.s3a.secret.key", config.awsSecretKey)
+                        .config("fs.s3a.endpoint", "s3.${manualConfig.regionName}.amazonaws.com")
+                        .config("spark.hadoop.fs.s3a.multiobjectdelete.enable","false")
+                        .config("spark.hadoop.fs.s3a.fast.upload","true")
+                        .config("spark.hadoop.mapreduce.fileoutputcommitter.algorithm.version", "2")
+                        .config("spark.speculation", "false")
+            }
+            return session.orCreate
+        }
+
+        @JvmStatic
+        fun convertToDataLakesIfPresent(integrationConfiguration: IntegrationConfiguration): Map<String, DataLake> {
+            val sourcesConfig = integrationConfiguration.datasources.orElse(listOf())
+            val destsConfig = integrationConfiguration.destinations.orElse(listOf())
+
+            // map to lakes if needed. This will be removed once launchpads are upgraded
+            return if ( !integrationConfiguration.datalakes.isPresent ){
```

Comment:
also super minor but I tend to prefer the built in functions like
```
return integrationConfiguration.datalakes.orElseGet {
  val newLakes = ArrayList<DataLake>()
  destsConfig.forEach { newLakes.add(it.asDataLake()) }
  sourcesConfig.forEach { newLakes.add(it.asDataLake()) }
  newLakes
}.associateBy { it.name }
```

Reply:
Yeah that definitely looks better

Reply:
Also looking at the code for `.map{}.toMap()` vs `associateBy`, `associateBy` wins by whatever a medium sized landslide is called